core/src/main/java/org/apache/calcite/statistic/MapSqlStatisticProvider.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.statistic;
+
+import org.apache.calcite.adapter.jdbc.JdbcTable;
+import org.apache.calcite.materialize.SqlStatisticProvider;
+import org.apache.calcite.plan.RelOptTable;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Implementation of {@link SqlStatisticProvider} that looks up values in a
+ * table.
+ *
+ * <p>Only for testing.
+ */
```

Comment:
We should move it to test source root if it is only used for testing.

Reply:
Not really possible. It is used outside of the core module.